src/realm/query_engine.hpp

```diff
@@ -1582,7 +1571,7 @@ class StringNode : public StringNodeBase {
         auto upper = case_map(v, true);
         auto lower = case_map(v, false);
         if (!upper || !lower) {
-            error_code = "Malformed UTF-8: " + std::string(v);
+            throw query_parser::InvalidQueryError(util::format("Malformed UTF-8: %1", v));
```

Comment:
These are not "query parser" errors, they are general query errors. Can we use another name?

Reply:
Ok - changed to std::runtime_error.